packages/protocol/contracts/governance/Election.sol

```diff
@@ -678,40 +716,64 @@ contract Election is
+    // Rounding may cause votesToUnits to return 0 for value != 0, preventing users
+    // from revoking the last of their votes. The case where value == votes is special cased
     // to prevent this.
-    uint256 unitsDelta = 0;
+    uint256 units = 0;
     uint256 activeVotes = getActiveVotesForGroupByAccount(group, account);
     GroupActiveVotes storage groupActive = active.forGroup[group];
     if (activeVotes == value) {
-      unitsDelta = groupActive.unitsByAccount[account];
+      units = groupActive.unitsByAccount[account];
     } else {
-      unitsDelta = getActiveVotesUnitsDelta(group, value);
+      units = votesToUnits(group, value);
     }
 
     groupActive.total = groupActive.total.sub(value);
-    groupActive.totalUnits = groupActive.totalUnits.sub(unitsDelta);
-    groupActive.unitsByAccount[account] = groupActive.unitsByAccount[account].sub(unitsDelta);
+    groupActive.totalUnits = groupActive.totalUnits.sub(units);
+    groupActive.unitsByAccount[account] = groupActive.unitsByAccount[account].sub(units);
+    return units;
   }
 
   /**
-   * @notice Returns the delta in active vote denominator for `group`.
+   * @notice Returns the number of units corresponding to `value` active votes.
    * @param group The address of the validator group.
-   * @param value The number of active votes being added.
-   * @return The delta in active vote denominator for `group`.
-   * @dev Preserves unitsDelta / totalUnits = value / total
+   * @param value The number of active votes.
+   * @return The corresponding number of units.
    */
-  function getActiveVotesUnitsDelta(address group, uint256 value) private view returns (uint256) {
+  function votesToUnits(address group, uint256 value) private view returns (uint256) {
+    FixidityLib.Fraction memory fixedValue = FixidityLib.newFixed(value);
     if (votes.active.forGroup[group].total == 0) {
-      return value;
+      return fixedValue.unwrap();
+    } else {
+      return
+        fixedValue.multiply(FixidityLib.wrap(votes.active.forGroup[group].totalUnits)).unwrap().div(
+          votes.active.forGroup[group].total
+        );
+    }
+  }
+
+  /**
+   * @notice Returns the number of active votes corresponding to `value` units.
+   * @param group The address of the validator group.
+   * @param value The number of units.
+   * @return The corresponding number of active votes.
+   */
+  function unitsToVotes(address group, uint256 value) private view returns (uint256) {
+    FixidityLib.Fraction memory fixedValue = FixidityLib.wrap(value);
+    if (votes.active.forGroup[group].totalUnits == 0) {
+      return 0;
     } else {
       return
-        value.mul(votes.active.forGroup[group].totalUnits).div(votes.active.forGroup[group].total);
+        fixedValue.multiply(FixidityLib.newFixed(votes.active.forGroup[group].total)).unwrap().div(
+          votes.active.forGroup[group].totalUnits
+        );
```

Comment:
Since units have more precision than votes, perhaps we could round to nearest here?
I think that the effect for this would be that when votes are added, that would be exact (when epoch rewards are paid, I don't think that can be exact, clearly it's possible to get half a vote as an epoch reward).

Reply:
Sorry I'm not totally clear on what you're suggesting?